PhysicsTools/PatAlgos/python/producersLayer1/lowPtElectronProducer_cff.py

```diff
@@ -69,11 +67,30 @@
                                                            genParticleMatch = "electronMatch"
                                                            )
 
-# Schedule rekeying of seed BDT ValueMaps by reco::GsfElectron for run2_miniAOD_UL
+# For run2_miniAOD_UL ...
 from Configuration.ProcessModifiers.run2_miniAOD_UL_cff import run2_miniAOD_UL
-from RecoEgamma.EgammaElectronProducers.lowPtGsfElectronSeedValueMaps_cff import rekeyLowPtGsfElectronSeedValueMaps
-from RecoEgamma.EgammaElectronProducers.lowPtGsfElectronID_cfi import lowPtGsfElectronID
 _makePatLowPtElectronsTask = makePatLowPtElectronsTask.copy()
+
+# (1) rekey seed BDT ValueMaps by reco::GsfElectron
+from RecoEgamma.EgammaElectronProducers.lowPtGsfElectronSeedValueMaps_cff import rekeyLowPtGsfElectronSeedValueMaps
 _makePatLowPtElectronsTask.add(rekeyLowPtGsfElectronSeedValueMaps)
+
+# (2) rerun ID
+from RecoEgamma.EgammaElectronProducers.lowPtGsfElectronID_cfi import lowPtGsfElectronID
 _makePatLowPtElectronsTask.add(lowPtGsfElectronID)
+
+# (3) apply energy regression
+from RecoEgamma.EgammaElectronProducers.lowPtGsfElectronFinalizer_cfi import lowPtGsfElectrons
+lowPtGsfElectronsPostRegression = lowPtGsfElectrons.clone(previousGsfElectronsTag = "lowPtGsfElectrons")
+_makePatLowPtElectronsTask.add(lowPtGsfElectronsPostRegression)
+
+# Switch to energy-regressed electrons as input
+sourceElectronsPostRegression = "lowPtGsfElectronsPostRegression"
+run2_miniAOD_UL.toModify(lowPtElectronMatch, src = sourceElectronsPostRegression)
+run2_miniAOD_UL.toModify(patLowPtElectrons, electronSource = sourceElectronsPostRegression)
+run2_miniAOD_UL.toModify(rekeyLowPtGsfElectronSeedValueMaps, gsfElectrons = sourceElectronsPostRegression)
+run2_miniAOD_UL.toModify(lowPtGsfElectronID, electrons = sourceElectronsPostRegression)
```

Comment:
```suggestion
_makePatLowPtElectronsTask.add(lowPtGsfElectrons)
```

if this is needed only in the context of run2_miniAOD_UL, why not simply use the same `lowPtGsfElectrons` module name?
you will need a `run2_miniAOD_UL.toModify(lowPtGsfElectrons, previousGsfElectronsTag = "lowPtGsfElectrons::@skipCurrentProcess")` in what is now `lowPtGsfElectronFinalizer_cfi`

Reply:
Done.